Exposures support for metrics in Core v1.4 (#2657)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/exposures.md

Exposures are defined in `.yml` files nested under an `exposures:` key.

<VersionBlock firstVersion="1.4">

<File name='models/<filename>.yml'>

```yaml
version: 2

exposures:

>   - name: weekly_jaffle_metrics

Reviewer: We added this in the "Example" section of `docs/reference/exposure-properties.md`, and I think it's a good idea to show off a new-ish capability added in v1.3:
```
label: Jaffles by the Week
```
(snippetize me, cap'n?)

Author: I updated both examples and added the new parts. I'll plot out a snippet and update for the final release

    type: dashboard
    maturity: high
    url: https://bi.tool/dashboards/1
    description: >
      Did someone say "exponential growth"?

    depends_on:
      - ref('fct_orders')
      - ref('dim_customers')
      - source('gsheets', 'goals')

>       - metric('example_metric_name')

Reviewer: Instead of `example_metric_name`, let's use [`count_orders`](https://github.com/dbt-labs/dbt-docs/blob/main/ci-project/models/metrics.yml#L5).

---

_big aside_

When writing pseudo code / example code, I try to coordinate all the code to use either:
1. "real" (imagined) names — most immersive, suggesting a potential use case
2. placeholder names (`example_metric_name`, `my_model`) — clearest & most flexible, though lacking the "why"
3. data structure (`<model_name: str>`) — most technical & informative, less accessible to less experienced users

This example uses "real" (imagined) names for models/sources/etc, loosely based on [`jaffle_shop`](https://github.com/dbt-labs/jaffle_shop) / the sample project [here](https://github.com/dbt-labs/dbt-docs/tree/main/ci-project).

    owner:

>       name: Claire from Data

Reviewer: At the risk of depriving Claire of some ongoing notoriety, we might swap this out for one of our newer cast of characters:
```suggestion
      name: Callum McData
```

</File>

</VersionBlock>

> ### Available properties

Reviewer 2: suggesting a table for the available properties, may help read easier

Author: Not opposed to it, but not sure how to present it in a way that makes sense. Open to ideas, but the testing I did looked a little cramped in table format.

Reviewer 2: makes sense!

</File>

</VersionBlock>

### Available properties

_Required:_

> - **name** (must be unique among exposures and you must use the [snake case](https://en.wikipedia.org/wiki/Snake_case) naming convention)

Reviewer 3: ```suggestion
- **name**: a unique exposure name written in [snake case](https://en.wikipedia.org/wiki/Snake_case)
```
